php/libraries/Database.class.inc

```diff
@@ -500,16 +517,26 @@ class Database
         $err = $this->trackChanges($table, array(), $where, 'D');
 
         $this->_printQuery($query);
-        $this->affected = $this->_PDO->exec($query);
-        if ($this->affected === false) {
+        $result = $this->_PDO->exec($query);
+        if ($result === false) {
             throw new DatabaseException(
                 "Database DELETE did not execute successfully."
                 . print_r($this->_PDO->errorInfo(), true),
                 $query,
                 $where
             );
-
+        } elseif ($result === 0) {
+            if (DEBUG) {
+                fwrite(
+                    STDERR,
+                    "Database DELETE executed successfully "
+                    . "but 0 matched rows: " . $query. "\n"
+                );
+                return;
+            }
         }
+
+        $this->affected = $result;
```

Comment:
Yeah I agree that fundamentally, it's just regular, valid SQL, but in our behavioral context of LORIS(I can't quite speak for the imaging side), I can't think of a scenario where we are issuing delete statements that should or happen to have nothing actually matched, so I think 1) there is some element of 'error', albeit possibly small 2) it should be made aware somehow and 3) code execution should continue.

Also, I agree $this->affected should still be 0 in those cases.
Right now, the code within my 
```
             if (DEBUG) {
                 ...
             }
```
blocks don't actually get executed - it was more of a temporary stub until we decide on something(I guess I should comment my code more often eh?) - so there is no premature 'return' before $this->affected gets set. This is because if DEBUG was true, there is another "if (DEBUG)" block a few lines above that already ends with a 'return'. Looking at it again, it really would have been a lot clearer if I left out the 'return' during my copy & paste of that chunk. But also like I mentioned, I think we should discuss this DEBUG constant. We have showdatabasequeries, sandbox tag, etc..